src/docfx/restore/RestoreMap.cs

```diff
@@ -1,69 +1,82 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Concurrent;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class RestoreMap
     {
-        private readonly RestoreLock _restoreLock;
+        private static readonly ConcurrentDictionary<string, string> s_mappings = new ConcurrentDictionary<string, string>();
```

Comment:
Since everything is static and readonly across a single process, can we expose 2 static methods to avoid passing objects around? like:

```csharp
Restore.GetGitRepositoryPath(string remote, string branch)
Restore.GetFileDownloadPath(string docsetPath, string pathOrUrl)
```

Reply:
not everything is static, we need keep the docset path context in every restore map, share it with source and loc docset

---
In reply to: [230308044](https://github.com/dotnet/docfx/pull/3570#discussion_r230308044) [](ancestors = 230308044)